app/addons/documents/doc-editor/components.react.jsx

```diff
@@ -0,0 +1,540 @@
+            <span>View Attachments</span>{' '}
+            <span className="caret"></span>
+          </button>
+          <ul className="dropdown-menu" role="menu" aria-labelledby="view-attachments-menu">
+            {this.getAttachmentList()}
+          </ul>
+        </div>
+      );
+    }
+  });
+
+
+  var PanelButton = React.createClass({
+    propTypes: {
+      title: React.PropTypes.string.isRequired,
+      onClick: React.PropTypes.func.isRequired
+    },
+
+    getDefaultProps: function () {
+      return {
+        title: '',
+        iconClass: '',
+        onClick: function () { }
+      };
+    },
+
+    render: function () {
+      var iconClasses = 'icon ' + this.props.iconClass;
+      return (
+        <div className="panel-section">
+          <button className="panel-button upload" title={this.props.title} onClick={this.props.onClick}>
+            <i className={iconClasses}></i>
+            <span>{this.props.title}</span>
+          </button>
+        </div>
+      );
+    }
+  });
+
+
+  var UploadModal = React.createClass({
+    propTypes: {
+      visible: React.PropTypes.bool.isRequired,
+      doc: React.PropTypes.object,
+      uploadComplete: React.PropTypes.func.isRequired
+    },
+
+    getInitialState: function () {
+      return {
+        inProgress: false,
+        loadPercentage: 0,
+        errorMessage: ''
+      };
+    },
+
+    componentDidUpdate: function () {
+      var params = (this.props.visible) ? { show: true, backdrop: 'static', keyboard: true } : 'hide';
+      $(this.getDOMNode()).modal(params);
+    },
+
+    // ensure that if the user clicks ESC to close the window, the store gets wind of it
+    componentDidMount: function () {
+      $(this.getDOMNode()).on('hidden.bs.modal', function () {
+        Actions.hideUploadModal();
+      });
+    },
+
+    componentWillUnmount: function () {
+      $(this.getDOMNode()).off('hidden.bs.modal');
+    },
+
+    closeModal: function () {
+      if (this.state.inProgress) {
+        Actions.cancelUpload();
+      }
+      Actions.hideUploadModal();
+
+      // timeout needed to only clear it once the animate close effect is done, otherwise the user sees it reset
+      // as it closes, which looks bad
+      setTimeout(function () {
+        this.reset();
+      }.bind(this), 1000);
+    },
+
+    reset: function () {
+      this.setState({
+        inProgress: false,
+        loadPercentage: 0,
+        errorMessage: ''
+      });
+      this.refs.uploadForm.getDOMNode().reset();
+    },
+
+    upload: function () {
+      if ($(this.refs.attachments.getDOMNode())[0].files.length === 0) {
+        this.setState({
+          errorMessage: 'Please select a file to be uploaded.'
+        });
+        return;
+      }
+
+      this.setState({
+        inProgress: true,
+        errorMessage: ''
+      });
+
+      Actions.uploadAttachment({
+        url: this.props.doc.url(),
+        $form: $(this.refs.uploadForm.getDOMNode()),
```

Comment:
Legacy! Good catch. The old code used $form + a plugin to handle the request generation. Removed.